stan/math/prim/err/check_positive_finite.hpp

```diff
@@ -17,15 +15,16 @@ namespace math {
  * @param name Variable name (for error messages)
  * @param y Variable to check
  * @throw <code>domain_error</code> if any element of y is not positive or
- *   if any element of y is NaN.
+ *   if any element of y is NaN or infinity.
  */
+
 template <typename T_y>
 inline void check_positive_finite(const char* function, const char* name,
                                   const T_y& y) {
-  check_positive(function, name, y);
-  check_finite(function, name, y);
+  auto is_good = [](const auto& y) { return y > 0 && std::isfinite(y); };
```

Comment:
Should this also be wrapped in an ```allFinite``` conditional?

Reply:
We have to check both conditions to do the screening thing, so I skipped this.